### PR TITLE
Fix duplicated packages when updating environment

### PR DESCRIPTION
**Pull request by gqmelo**
When updating an environment, the currently installed packages were not
being considered, so new versions of packages were installed without
uninstalling old versions.

### CODE REVIEW

File: tests/conda_env/test_create.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function
```
Comment (gqmelo) [inline tests/conda_env/test_create.py]: This file was based on conda's test_create.py
Please let me know if this should be different

**Comment by gqmelo** [comment]
This fix was ported from https://github.com/conda/conda-env/pull/274 and seems to be related to https://github.com/conda/conda/issues/3478

**Comment by gqmelo** [comment]
This test is failing on Windows, but I don't think it is related to my changes:

```python
=================================== FAILURES ===================================
_______________ ExportIntegrationTests.test_multi_channel_export _______________
Traceback (most recent call last):
  File "/home/travis/build/conda/conda/tests/test_export.py", line 50, in test_multi_channel_export
    self.assertIn("conda-forge", output)
  File "/opt/python/2.7.9/lib/python2.7/unittest/case.py", line 803, in assertIn
    self.fail(self._formatMessage(msg, standardMsg))
  File "/opt/python/2.7.9/lib/python2.7/unittest/case.py", line 410, in fail
    raise self.failureException(msg)

AssertionError: 'conda-forge' not found in '# This file may be used to create an environment using:\n# $ conda create --name <env> --file <this file>\n# platform: linux-64\nopenssl=1.0.2j=0\npip=8.1.2=py35_0\npython=3.5.2=0\nreadline=6.2=2\nsetuptools=27.2.0=py35_0\nsix=1.10.0=py35_0\nsqlite=3.13.0=0\ntk=8.5.18=0\nwheel=0.29.0=py35_0\nxz=5.2.2=0\nzlib=1.2.8=3\n'
```

**Comment by kalefranz** [comment]
Good PR.  And thanks.  Will merge before the next 4.2.x release.

**Pull request merged**